client-lite/src/config/mcc_manager.cpp

```diff
@@ -50,14 +50,18 @@ boost::optional<std::chrono::seconds> MCCManager::FallbackDelay()
 
 std::string MCCManager::GetHost()
 {
-    std::string mccHostName =_configManager.CacheHostServer();
-    if (mccHostName.empty())
+    boost::optional<std::string> mccHostNameOpt =_configManager.CacheHostServer();
+    std::string mccHostName;
+
+    if (!mccHostNameOpt.is_initialized())
```

Comment:
>!mccHostNameOpt.is_initialized() [](http://example.com/codeflow?start=8&length=32)

Flip the conditional to avoid the negation (improves readability):
if (mccHostNameOpt.is_initialized())
{
    mccHostName = mccHostNameOpt.get();
}
else
{
    ...
}